vm/src/stdlib/ctypes/array.rs

```diff
@@ -0,0 +1,633 @@
+                    Err(vm.new_index_error("index out of bounds".to_string()))
+                } else {
+                    let idx = idx as usize;
+                    let buffer_slice = buffer_bytes[idx..idx + offset].as_ref();
+                    Ok(byte_to_pyobj(zelf._type_._type_.as_str(), buffer_slice, vm))
+                }?
+            }
+            Either::B(slice) => {
+                let (step, mut start, mut stop) = array_get_slice_inner(slice, vm)?;
+
+                let slice_length =
+                    slice_adjust_size(zelf._length_ as isize, &mut start, &mut stop, step) as usize;
+
+                let mut obj_vec = Vec::with_capacity(slice_length);
+
+                for i in (start as usize..stop as usize).step_by(step as usize) {
+                    obj_vec.push(byte_to_pyobj(
+                        zelf._type_._type_.as_str(),
+                        buffer_bytes[i..i + offset].as_ref(),
+                        vm,
+                    ));
+                }
+
+                PyList::from(obj_vec).into_object(vm)
+            }
+        };
+
+        Ok(res)
+    }
+
+    #[pymethod(magic)]
+    fn setitem(
+        zelf: PyRef<Self>,
+        k_or_idx: Either<isize, PyRef<PySlice>>,
+        obj: PyObjectRef,
+        vm: &VirtualMachine,
+    ) -> PyResult<()> {
+        let buffer = try_buffer_from_object(vm, zelf.as_object())?;
+        let buffer_size = buffer.get_options().len;
+        let mut buffer_bytes = buffer.obj_bytes_mut();
+
+        let offset = buffer_size / zelf.len();
+
+        match k_or_idx {
+            Either::A(idx) => {
+                if idx < 0 {
+                    Err(vm.new_index_error("invalid index".to_string()))
+                } else if idx as usize > zelf._length_ {
+                    Err(vm.new_index_error("index out of bounds".to_string()))
+                } else {
+                    set_array_value(&zelf, &mut buffer_bytes, idx as usize, offset, obj, vm)
+                }
+            }
+            Either::B(slice) => {
+                let (step, mut start, mut stop) = array_get_slice_inner(slice, vm)?;
+
+                let slice_length =
+                    slice_adjust_size(zelf._length_ as isize, &mut start, &mut stop, step) as usize;
+
+                let values: Vec<PyObjectRef> = vm.extract_elements(&obj)?;
+
+                if values.len() != slice_length {
+                    Err(vm.new_value_error("can only assign sequence of same size".to_string()))
+                } else {
+                    let mut cur = start as usize;
+
+                    for v in values {
+                        set_array_value(&zelf, &mut buffer_bytes, cur, offset, v, vm)?;
+                        cur += step as usize;
+                    }
+                    Ok(())
+                }
+            }
+        }
+    }
+}
+
+impl PyCDataFunctions for PyCArray {
+    fn size_of_instances(zelf: PyRef<Self>, vm: &VirtualMachine) -> PyResult<PyObjectRef> {
```

Comment:
```suggestion
    fn size_of_instances(zelf: PyRef<Self>, vm: &VirtualMachine) -> PyResult<usize> {
```

Reply:
Will `usize` be converted to a Python `int`?

Reply:
Yes, when it eventually returned for `#[pyfunction]`, `#[pymethod]` or `#[pyproperty]`.